src/main/kotlin/com/bugsnag/android/gradle/BugsnagGenerateUnitySoMappingTask.kt

```diff
@@ -48,19 +55,135 @@ internal open class BugsnagGenerateUnitySoMappingTask @Inject constructor(
     val intermediateOutputDir: DirectoryProperty = objects.directoryProperty()
         .convention(projectLayout.buildDirectory.dir(UNITY_SO_MAPPING_DIR))
 
+    @get:OutputDirectory
+    val unitySharedObjectDir: DirectoryProperty = objects.directoryProperty()
+        .convention(projectLayout.buildDirectory.dir(UNITY_SO_COPY_DIR))
+
+    @get:InputDirectory
+    val rootProjectDir: DirectoryProperty = objects.directoryProperty()
+
     @TaskAction
     fun generateMappingFiles() {
         logger.lifecycle("Generating Unity mapping files")
-        // TODO generate mapping file here
+
+        // TODO search Unity internal build too
+        val symbolArchives = getUnitySymbolArchives(rootProjectDir)
+        logger.info("Found symbol archives: $symbolArchives")
```

Comment:
This might be confusing if it's empty and so warns immediately afterwards — maybe it'd be better in an `else` to the if statement below?

Reply:
Moved to an `else` statement